packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl

```diff
@@ -1,12 +1,26 @@
 import 'dart:async';
 
 import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show visibleForTesting;
 
 class {{pluginDartClass}} {
-  static const MethodChannel _channel =
-      const MethodChannel('{{projectName}}');
+  factory {{pluginDartClass}}() {
```

Comment:
I don't really think this is necessary to make the channel testable. There is already a method on `MethodChannel.setMockMethodCallHandler` for testing the dart side

Reply:
thank you for your suggestion, I was following example from https://github.com/flutter/plugins/blob/master/packages/battery/lib/battery.dart#L14 . Let me do further research and  resolve this.

Reply:
resolve in latest commit